h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -1738,7 +1737,8 @@ public int getCurrentRowSize() {
           } else if (v.isUUID()) sb.append(PrettyPrint.UUID(_curChks[i].at16l(_chkRow), _curChks[i].at16h(_chkRow)));
           else if (v.isInt()) sb.append(_curChks[i].at8(_chkRow));
           else if (v.isString()) {
-            final String escapedString = escapeQuotesForCsv(_curChks[i].atStr(tmpStr, _chkRow).toString());
+            final BufferedString unescapedTempStr = new BufferedString();
```

Comment:
any reason to move it here? I think the reason this was declared outside was to avoid repeated allocation

Reply:
This is a common premature optimization mistake many Java devs believe for whatever reason. Javac is clever enough to optimize this. So I followed the rule of narrowest variable scope possible. I can move it outside if this is a problem for whatever reason. 

https://stackoverflow.com/questions/8803674/declaring-variables-inside-or-outside-of-a-loop

Reply:
@Pscheidl this is not true. There is a difference between **declaring** and creating a new instance.

Please keep in mind that I am not arguing whether your change is good/makes an impact or not.

In your modified code the instance will be created per-column (chunk). The BufferedString was designed to be mutable and save you from new object creation. The `atStr` method just updates the pointers on the existing object, no new memory is allocated. This allows for faster iteration over String columns. Obviously the benefits go away as long as you call `toString`.

BTW method getBytesForRow is a huge performance bottleneck, it is critical for good export performance and I have a pending improvement: https://github.com/h2oai/h2o-3/pull/4913/files#diff-ee1c3497b4c32b3ec0b3577c3d1e736b28bac51e97841cb15778610a9255072cR1712

Reply:
First, I confused `BufferedString` and `StringBuffer`. Didn't think it was possible to reuse it and didn't look properly in the code. Yes, after looking into `atStr_impl`, I can see the bytes copied anyway. Didn't notice the contract. I was convince we initialize it every time anyway.

Reverted in https://github.com/h2oai/h2o-3/pull/5243/commits/1cb0c48822a4fe7e2e71126fd45f92591f5a798b.